app/src/main/java/org/fossasia/openevent/general/MainActivity.kt

```diff
@@ -68,10 +69,23 @@ class MainActivity : AppCompatActivity() {
             }
             R.id.orderCompletedFragment -> navController.popBackStack(R.id.eventDetailsFragment, false)
             R.id.welcomeFragment -> finish()
+            R.id.editProfileFragment -> {
+
+                // Calls the handleBackPress method in EditProfileFragment
+                val hostFragment = supportFragmentManager.findFragmentById(R.id.frameContainer) as? NavHostFragment
+                (hostFragment?.childFragmentManager?.fragments?.get(0) as? EditProfileFragment)?.handleBackPress()
```

Comment:
Too complex.

Reply:
Should I split these up into more lines of code or do you mean some other way should be used? Because I could not find any other way

Reply:
You just need to call the method in edit profile fragment. Can't you do it by taking the fragment as object?

Reply:
How do I get a reference to the Fragment instance?